src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -161,9 +163,14 @@ void DnsStreamHandler::process_udp_packet_cb(pcpp::Packet &payload, PacketDirect
         metric_port = dst_port;
     }
     if (metric_port) {
-        DnsLayer dnsLayer(udpLayer, &payload);
-        if (!_filtering(dnsLayer, dir, l3, pcpp::UDP, metric_port, stamp)) {
-            _metrics->process_dns_layer(dnsLayer, dir, l3, pcpp::UDP, flowkey, metric_port, stamp);
+        if (flowkey != cached_dns_layer.flowKey || stamp.tv_sec != cached_dns_layer.timestamp.tv_sec || stamp.tv_nsec != cached_dns_layer.timestamp.tv_nsec) {
```

Comment:
@weyrick  using only packet timestamp still not unique enough. IT happens to have packets with same timestamp at least in pktvisor tests. So I need to use both FlowKey AND timestamp (sec and nsec)